crates/red_knot_python_semantic/src/types.rs

```diff
@@ -6231,10 +6231,20 @@ impl<'db> IntersectionType<'db> {
             elements
         }
 
+        let normalised_negative = normalized_set(db, self.negative(db));
+
+        if self.positive(db).is_empty() {
+            return IntersectionType::new(
+                db,
+                FxOrderSet::from_iter([Type::object(db)]),
+                normalised_negative,
+            );
+        }
```

Comment:
It seems like we shouldn't need this change, if `IntersectionBuilder` always ensures it already.